### PR TITLE
Phase2 bughunt v7

### PR DESCRIPTION
**Pull request by mayank2913**
- [x] JSON edits not working properly fixed
- [x] Data type for synth_id being shown as blank in column edit mode
- [x] Downloaded file name after save session and review changes fixed
- [x] Deletion of the primary key column from the columns tab in the table should remove the primary key 
- [x] Issue with Interleave Fixed
- [x] Two columns can be renamed to the same column and saved. This should not be allowed.
- [x] UI is allowing two indexes to be created with same name.

### CODE REVIEW

File: spanner/ddl/ast.go [hunk 1]
```diff
@@ -292,6 +292,9 @@ func (ct CreateTable) PrintCreateTable(config Config) string {
 		}
 	}
 
+	if len(keys) == 0 {
+		return fmt.Sprintf("%sCREATE TABLE %s (\n%s) %s", tableComment, config.quote(ct.Name), cols, interleave)
```
Comment (shreyakhajanchi) [inline spanner/ddl/ast.go:296]: getting syntax error due to extra `,` : Error parsing Spanner DDL statement: CREATE TABLE `test123` ( `name` STRING(32), `age` INT64, ) : Syntax error on line 4, column 2: Expecting 'PRIMARY' but found 'EOF'

File: webv2/typemap/mysql_typemap.go [hunk 2]
```diff
@@ -125,10 +125,10 @@ func ToSpannerTypeMySQL(srcType string, spType string, mods []int64) (ddl.Type,
 		return ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, nil
 	case "json":
 		switch spType {
-		case ddl.Bytes:
```
Comment (shreyakhajanchi) [inline webv2/typemap/mysql_typemap.go:128]: why are we removing this byte case?

File: webv2/typemap/postgres_typemap.go [hunk 3]
```diff
@@ -152,6 +152,13 @@ func ToSpannerTypePostgres(srcType string, spType string, mods []int64) (ddl.Typ
 		default:
 			return ddl.Type{Name: ddl.Timestamp}, []internal.SchemaIssue{internal.Timestamp}
 		}
+	case "json":
```
Comment (shreyakhajanchi) [inline webv2/typemap/postgres_typemap.go:155]: add jsonb as well as in postgres/toddl.go

File: webv2/typemap/sqlserver_typemap.go [hunk 4]
```diff
@@ -81,6 +81,13 @@ func ToSpannerTypeSQLserver(srcType string, spType string, mods []int64) (ddl.Ty
 			}
 			return ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, nil
 		}
+	case "json":
```
Comment (shreyakhajanchi) [inline webv2/typemap/sqlserver_typemap.go:84]: json datatype is not present in sql server

File: webv2/session/session_handler.go [hunk 5]
```diff
@@ -230,6 +230,14 @@ func SaveRemoteSession(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	sessionMetaData := GetSessionState().SessionMetadata
+	
+	sessionMetaData.DatabaseName = sm.DatabaseName
+	sessionMetaData.DatabaseType = sm.DatabaseType
+	sessionMetaData.SessionName = sm.SessionName
+
+	GetSessionState().SessionMetadata = sessionMetaData
```
Comment (manitgupta) [inline webv2/session/session_handler.go:239]: This looks weird 🤔
Why create a the temp variable `sessionMetaData`?
You can just do `GetSessionState().SessionMetadata.DatabaseName = sm.DatabaseName`

File: webv2/typemap/postgres_typemap.go (hunk 3 above)
Comment (Ronak-Searce) [reply webv2/typemap/postgres_typemap.go:155]: Updated

File: webv2/typemap/sqlserver_typemap.go (hunk 4 above)
Comment (Ronak-Searce) [reply webv2/typemap/sqlserver_typemap.go:84]: Updated

File: webv2/typemap/mysql_typemap.go (hunk 2 above)
Comment (mayank2913) [reply webv2/typemap/mysql_typemap.go:128]: Updated

File: ui/src/app/components/object-detail/object-detail.component.ts [hunk 6]
```diff
@@ -314,6 +314,7 @@ export class ObjectDetailComponent implements OnInit {
             data: { message: res, type: 'error' },
             maxWidth: '500px',
           })
+          this.isEditMode = true
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.ts:317]: Interleaving suggestion doesn't go away if you delete pk column and then fk : https://drive.google.com/file/d/1KZP9ow_RFnGlTl8K2Mq7EK9_RpzJzRqw/view?usp=share_link&resourcekey=0-I4iwAqWiZynbMp3FMOhLsA

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 7]
```diff
@@ -225,7 +225,7 @@ <h3 class="title">
             <ng-container matColumnDef="spColName">
               <th mat-header-cell class="table_header" *matHeaderCellDef>Name</th>
               <td mat-cell *matCellDef="let element">
-                <div *ngIf="isEditMode && element.get('spOrder').value !== ''">
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:228]: Still getting undefined_undefined_undefined.json after saving session then resuming and then interleaving the table

File: ui/src/app/components/object-detail/object-detail.component.ts (hunk 6 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.ts:317]: Updated

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 7 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.html:228]: Updated

**Review by shreyakhajanchi — COMMENTED** [review]
LGTM

**Review by shreyakhajanchi — APPROVED** [review]

**Pull request merged**